ReactWindows/ReactNative.Shared/ReactInstanceManager.cs

```diff
@@ -171,23 +171,37 @@ public ReactContext CurrentReactContext
                 }
                 _hasStartedCreatingInitialContext = true;
 
-                return await CreateReactContextCoreAsync(token);
+                ReactContext context = null;
+                try
+                {
+                    context = await CreateReactContextCoreAsync(token);
+                }
+                finally
+                {
+                    if (context == null)
+                    {
+                        _hasStartedCreatingInitialContext = false;
+                    }
+                }
+
+                return context;
             }
         }
 
         /// <summary>
         /// Awaits the currently initializing React context.
         /// </summary>
         /// <param name="token">A token to cancel the request.</param>
+        /// <param name="noThrow">If true, method returns null instead of throwing if no context exists.</param>
         /// <returns>
         /// A task to await the React context.
         /// </returns>
-        public async Task<ReactContext> GetReactContextAsync(CancellationToken token)
+        public async Task<ReactContext> GetReactContextAsync(CancellationToken token, bool noThrow = false)
```

Comment:
afaic default args are discouraged in favor of overloads (or at least some of the checks flag these)

Reply:
Agree - let's avoid default args on the public surface. Honestly, I'd be more in favor of just making a slight breaking change here and not throwing at all. I imagine the equivalent of what you are trying to check here is `_hasStartedCreatingInitialContext == false`. On Android, they actually expose this value: https://github.com/facebook/react-native/blob/master/ReactAndroid/src/main/java/com/facebook/react/ReactInstanceManager.java#L405-L411

Reply:
Ok, I'll follow what Android does if you thing a slight breaking change is ok.

---
In reply to: [189276480](https://github.com/Microsoft/react-native-windows/pull/1827#discussion_r189276480) [](ancestors = 189276480)

Reply:
How will it be breaking to add a new property?

Reply:
I may have mix up with some other Q. I'll do my best :)

---
In reply to: [189310470](https://github.com/Microsoft/react-native-windows/pull/1827#discussion_r189310470) [](ancestors = 189310470)

Reply:
Ok, I don't really like that Android way, we do have the flag protected by the awaitable lock, to allow for simultaneous asynchronous initializations. So if you don't want a default param I'll just add a TryGetBlah method that doesn't throw

---
In reply to: [189311202](https://github.com/Microsoft/react-native-windows/pull/1827#discussion_r189311202) [](ancestors = 189311202,189310470)

Reply:
@rozele Pushed the change.

---
In reply to: [189313217](https://github.com/Microsoft/react-native-windows/pull/1827#discussion_r189313217) [](ancestors = 189313217,189311202,189310470)

Reply:
Great - just delete the default arg and LGTM.

Reply:
Oh boy, too in a hurry...

---
In reply to: [189319231](https://github.com/Microsoft/react-native-windows/pull/1827#discussion_r189319231) [](ancestors = 189319231)